eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jdbc/JdbcEventStorageEngine.java

```diff
@@ -301,15 +302,12 @@ protected PreparedStatement appendSnapshot(Connection connection, DomainEventMes
      * @param connection          The connection to the database.
      * @param aggregateIdentifier The identifier of the aggregate whose snapshots to delete.
      * @return A {@link PreparedStatement} that deletes all the aggregate's snapshots when executed.
-     *
      * @throws SQLException when an exception occurs while creating the prepared statement.
      */
     protected PreparedStatement deleteSnapshots(Connection connection, String aggregateIdentifier, long sequenceNumber)
             throws SQLException {
-        PreparedStatement preparedStatement = connection.prepareStatement(
-                "DELETE FROM " + schema.snapshotTable() + " WHERE " + schema.aggregateIdentifierColumn() + " = ? "
-                        + "AND " + schema.sequenceNumberColumn() + " < ?"
-        );
+        final String sql = schema.deleteSnapshotsSqlStatement();
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
         preparedStatement.setString(1, aggregateIdentifier);
```

Comment:
Hmmm... This pretty tightly couples SQL string with the prepared statement in terms of parameters. I'd go with approach that `schema` creates the statement with given parameters (`aggregateIdentifier`, and `sequenceNumber` in this case). What do you think?

Reply:
Changes are done as discussed.